mindmeld/models/nn_utils/input_encoders.py

```diff
@@ -0,0 +1,937 @@
+        """
+
+        examples, labels = self._reformat_and_validate(examples, labels)
+
+        # convert to tokens and obtain sequence lengths
+        examples = [self._tokenizer(text) for text in examples]
+        sequence_lengths = [len(seq) for seq in examples]
+
+        # if padding_length is None, it is computed as max(length of all seqs from inputted text)
+        padding_length = padding_length or self.padding_length
+        # add 2 for start and end tokens respectively if add_terminals is True
+        add_terminals = add_terminals or self.add_terminals
+        curr_max = max(sequence_lengths) + 2 if add_terminals else max(sequence_lengths)
+        padding_length = min(padding_length, curr_max) if padding_length else curr_max
+
+        # batchify by truncating or padding for each example
+        seq_ids, sequence_lengths = zip(*[
+            self.encode(example, padding_length, add_terminals) for example in examples
+        ])
+
+        return_dict = {
+            "seq_ids": torch.as_tensor(seq_ids, dtype=torch.long),
+            "seq_lengths": torch.as_tensor(sequence_lengths, dtype=torch.long)
+        }
+
+        if _return_tokenized_examples:
+            return_dict.update({
+                "seq_tokens": examples
+            })
+
+        if labels:
+            new_labels = [self.encode_labels(label, padding_length, add_terminals) for label in
+                          labels]
+            return_dict.update({
+                "labels": torch.as_tensor(new_labels, dtype=torch.long)
+            })
+
+        return return_dict
+
+
+class SeqClsEncoderWithPlmLayer(PretrainedLangModelBasedEncoder):
+
+    def batch_encode(
+        self,
+        examples: Union[str, List[str]],
+        labels: Union[int, List[int]] = None,
+        padding_length: int = None,
+    ):
+        """
+        Returns batched encodings that can be used as an input to a featurizer
+        """
+
+        examples, labels = self._reformat_and_validate(examples, labels)
+
+        return_dict = {}
+        padding_length = padding_length or self.padding_length
+
+        if self.embedder_type == "bert":
+            # If padding_length is None, padding is done to the max length of the input batch
+            _inputs = self._bert_tokenizer(
+                examples, padding=True, truncation=True, max_length=padding_length,
+                return_tensors="pt"
+            ).to(self.device)
+            _outputs = self._bert_model(**_inputs, return_dict=True)
+
+            # last_hidden_state, pooler_output, seq_lengths are the keys being updated
+            return_dict.update({
+                **_outputs,
+                "seq_lengths": _inputs["attention_mask"].sum(dim=-1)
+            })
+
+        if labels:
+            return_dict.update({"labels": torch.as_tensor(labels, dtype=torch.long)})
+
+        return return_dict
+
+
+class TokenClsEncoderWithPlmLayer(PretrainedLangModelBasedEncoder):
```

Comment:
`TokenClsEncoderWithPLMLayer`

Reply:
More docs here please. Its not clear what PLM is here. Also expand the name